operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -135,8 +134,67 @@ func deprecatedSettingsMessage(iop *v1alpha1.IstioOperatorSpec) string {
 }
 
 // validateFeatures check whether the config sematically make sense. For example, feature X and feature Y can't be enabled together.
-func validateFeatures(_ *valuesv1alpha1.Values, _ *v1alpha1.IstioOperatorSpec) util.Errors {
-	return nil
+func validateFeatures(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
+	return CheckServicePorts(values, spec)
+}
+
+func CheckServicePorts(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
+	var errs util.Errors
+	if !values.GetGateways().GetIstioIngressgateway().GetRunAsRoot().GetValue() {
```

Comment:
this long chain of derefs looks worrisome here and below - are we sure there's no null here? if so a comment would be useful about the assumptions.

Reply:
Get...() is null safe in proto. The functions all look like
```
GetDestination() *Destination {
	if m != nil {
		return m.Destination
	}
	return nil
}
```